pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1004,7 +1038,10 @@ impl LayerInner {
             //
             // FIXME: this is not true anymore, we can safely evict wanted deleted files.
         } else if can_evict && evict {
-            let span = tracing::info_span!(parent: None, "layer_evict", tenant_id = %self.desc.tenant_shard_id.tenant_id, shard_id = %self.desc.tenant_shard_id.shard_slug(), timeline_id = %self.desc.timeline_id, layer=%self, %version);
+            // If timeline is alive, we can construct a span with IDs for this function.
+            let span = self.timeline.upgrade().map(|timeline| {
```

Comment:
@koivunej at this point, we could instead `return` if the timeline is gone.  I didn't make that change because it wasn't obvious if that would violate any other expectations, but it's _probably_ fine since once Timeline is destroyed we don't have any obligation to make progress with eviction -- your thoughts?